File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/loader/LocalFSStoredItem.java
package ru.kmorozov.gbd.core.loader;

import ru.kmorozov.gbd.core.logic.model.book.base.IPage;

import java.io.File;

public class LocalFSStoredItem extends RawFileItem {

    protected final LocalFSStorage storage;
    protected final IPage page;
    protected final String imgFormat;

    protected LocalFSStoredItem(File file, LocalFSStorage storage, IPage page, String imgFormat) {
        super(file);

        this.storage = storage;
        this.page = page;
        this.imgFormat = imgFormat;
    }

    public LocalFSStoredItem(LocalFSStorage storage, IPage page, String imgFormat) {
        this(new File(storage.getStorageDir().getPath() + File.separator + page.getOrder() + '_' + page.getPid() + '.' + imgFormat), storage, page, imgFormat);
    }
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/loader/LocalFSStoredItem.kt
package ru.kmorozov.gbd.core.loader

import ru.kmorozov.gbd.core.logic.model.book.base.IPage

import java.io.File

open class LocalFSStoredItem : RawFileItem {

    protected val storage: LocalFSStorage
    protected val page: IPage
    protected val imgFormat: String

    protected constructor(file: File, storage: LocalFSStorage, page: IPage, imgFormat: String) : super(file) {
        this.storage = storage
        this.page = page
        this.imgFormat = imgFormat
    }

    constructor(storage: LocalFSStorage, page: IPage, imgFormat: String) : this(File(storage.storageDir.path + File.separator + page.order + '_'.toString() + page.pid + '.'.toString() + imgFormat), storage, page, imgFormat) {}
}
